invenio_admin/ext.py

```diff
@@ -26,25 +26,39 @@
 
 from __future__ import absolute_import, print_function
 
-from flask_babelex import gettext as _
+import pkg_resources
+from flask_admin import Admin
 
-from .views import blueprint
+from . import config
+from .views import ProtectedAdminIndexView, blueprint
 
 
 class InvenioAdmin(object):
     """Invenio-Admin extension."""
 
-    def __init__(self, app=None):
+    def __init__(self, app=None, **kwargs):
         """Extension initialization."""
-        _('A translation string')
         if app:
-            self.init_app(app)
+            self.init_app(app, **kwargs)
 
-    def init_app(self, app):
+    def register(self, register_call):
+        """Register the admin views within register_call to current admin."""
+        register_call(self._admin)
+
+    def init_app(self, app, entrypoint_name='invenio_admin.views',
+                 **kwargs):
         """Flask application initialization."""
         self.init_config(app)
         app.register_blueprint(blueprint)
         app.extensions['invenio-admin'] = self
+        admin = Admin(app, name=app.config['ADMIN_APPNAME'],
```

Comment:
IMHO `self.admin` is fine. Is there a reason why do you want to *protect* the `Admin` instance?